src/Symfony/Component/Form/ChoiceList/Factory/Cache/AbstractStaticOption.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\ChoiceList\Factory\Cache;
+
+use Symfony\Component\Form\ChoiceList\Factory\CachingFactoryDecorator;
+use Symfony\Component\Form\ChoiceList\Loader\ChoiceLoaderInterface;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\FormTypeExtensionInterface;
+use Symfony\Component\Form\FormTypeInterface;
+
+/**
+ * A template decorator for static {@see ChoiceType} options.
+ *
+ * Used as fly weight for {@see CachingFactoryDecorator}.
+ *
+ * @internal
+ *
+ * @author Jules Pietri <jules@heahprod.com>
+ */
+abstract class AbstractStaticOption
+{
+    private static $options = [];
+
+    /** @var bool|callable|string|array|\Closure|ChoiceLoaderInterface */
+    private $option;
+
+    /**
+     * @param FormTypeInterface|FormTypeExtensionInterface $formType A form type or type extension configuring a cacheable choice list
+     * @param mixed                                        $option   Any pseudo callable, array, string or bool to define a choice list option
+     * @param mixed|null                                   $vary     Dynamic data used to compute a unique hash when caching the option
+     */
+    final public function __construct($formType, $option, $vary = null)
```

Comment:
why final?

Reply:
Why not? This class is responsible for clearing the cache of all options, they should not override/extends it. I see no use case currently, we have everything we need here with those  three args.

Reply:
This is a design needed by the `CachingFactoryDecorator` no matter the option.